src/core/types.h

```diff
@@ -788,7 +788,8 @@ int binarySearch(T needle, RandomAccessIterator start, RandomAccessIterator end,
     int l = 0;
     int r = end - start - 1;
     while (l <= r) {
-        int mid = l + (r - l) / 2;
```

Comment:
For BinarySearch, 
      0 <= l <= r 
--> 0 <= r-l <= r
--> 0 <= (r-l)/2 <= r/2, it won't overflow.

0 <= l <= l + (r-l)/2 = (l + r)/2 <= r, it won't overflow.
As @corona10 you mentioned, if needle value is right-end near value, a bigger L will be reached when needle value is bigger than the maximum value (for calculating middle position, L will be equal to r), even in that case, (r-l)/2 will still not overflow, l + (r-l)/2 also won't overflow.

Jianbin Wang